src/minishell.c

```diff
@@ -96,6 +82,7 @@ int	main(int argc, char *argv[], char *envp[])
 		ft_putstr_fd("Invalid arguments. Try ./minishell\n", 2);
 	signal(SIGINT, sighandler);
 	signal(SIGQUIT, SIG_IGN);
+	lst = env_to_dll(envp);
```

Comment:
I've run a few tests with this and haven't found any particular problems yet, just found your parse to be amazing.